interview-questions.md

```diff
@@ -7,60 +7,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is an instance variable in Ruby? How is it different from other variables?
 
-  Your answer:
+  Your answer: The instance variable in Ruby means a variable belongs to the instance of a class. This is different from other variables because it is only used in Ruby classes.w
 
-  Researched answer:
+  Researched answer: You can't just access an instance variable from outside of an object, as you can with variables in JavaScript. Think local scope.
 
 
 
 2. What is a block in Ruby?
 
-  Your answer:
+  Your answer: A block in Ruby is a piece of code that accepts an argument and returns a value. 
 
-  Researched answer:
+  Researched answer:A block is the same thing as a method, but it does not belong to an object. A block is always invoked with a function or can say passed to a method call.
 
 
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: Return does not have to be called for you to recieve an output.
 
-  Researched answer:
+  Researched answer: Ruby returns the value of the last executed instruction in the method. In the implicit_return method, as if true is always evaluated as true (mister obvious) then the last executed instruction is 42.
 
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: Object-oriented programming relies on classes and objects.
 
-  Researched answer:
+  Researched answer:In functional programming, data cannot be stored in objects and it can only be transformed by creating functions. In object-oriented programming, data is stored in objects. 
```

Comment:
Close, Functional programming makes a clear distinction between data and behavior. Behavior is not stored in objects like it is with OOP but instead is a construct of the functional application.